Extend subgraph

bsamuels453 opened the pull request:
https://github.com/yearn/thegraph-yfi-buyback/issues/1

also had to upgrade graph-cli/graph-ts because 0.18.0 isn't available for my distro

rareweasel commented on schema.graphql:
```diff
@@ -8,3 +8,12 @@ type BuyBack @entity {
   yfi: BigInt!
 }
+
+type BuyBackContract @entity {
```
Question:

I wonder if we need to store history (and tx hash for tracking purposes). I guess it not required, but just in case.

rareweasel reviewed (approved):
LGTM (I've just left a comment/question), but maybe @milkyklim has more context.

milkyklim commented:
Instead of 2 separate entries can you create one: `Admin`? Update it depending on the event you get. 
```
Admin {
  address 
  block 
  timestamp 
  hash
  id
}
```
Same for treasury – `Treasury`. 

This should simplify the codebase quite a bit. @bsamuels453

bsamuels453 commented:
@milkyklim let me know if this is closer to what you're looking for

milkyklim commented:
Looks better! Any reason you keep id=1 for all new incoming entries? Prefer to use time travel queries when you need to pull previous treasury/admin?

bsamuels453 commented:
> Looks better! Any reason you keep id=1 for all new incoming entries? Prefer to use time travel queries when you need to pull previous treasury/admin?

wasn't aware there was an interest in tracking historical. updated the commit. If you need to query the current admin/treasurer, use the conditional `limit:1, orderBy: block; orderDirection: desc`

also added daiRemaining to BuyBack for keeping track of dai balance
